// @thirdweb-dev/wallets/evm/dist/thirdweb-dev-wallets-evm.esm.js
export { E as EIP155_SIGNING_METHODS } from '../../dist/wc-c5b5676b.esm.js';
export { w as walletIds } from '../../dist/walletIds-f8212319.esm.js';
export { a as AbstractClientWallet, D as DEFAULT_DAPP_META } from '../../dist/base-3454e9e0.esm.js';
export { C as Connector, W as WagmiAdapter } from '../../dist/connector-20f7cf73.esm.js';
export { PaperWallet } from '../wallets/paper-wallet/dist/thirdweb-dev-wallets-evm-wallets-paper-wallet.esm.js';
export { CoinbaseWallet } from '../wallets/coinbase-wallet/dist/thirdweb-dev-wallets-evm-wallets-coinbase-wallet.esm.js';
export { LocalWallet, isValidPrivateKey } from '../wallets/local-wallet/dist/thirdweb-dev-wallets-evm-wallets-local-wallet.esm.js';
export { FrameWallet } from '../wallets/frame/dist/thirdweb-dev-wallets-evm-wallets-frame.esm.js';
export { InjectedWallet } from '../wallets/injected/dist/thirdweb-dev-wallets-evm-wallets-injected.esm.js';
export { MetaMaskWallet } from '../wallets/metamask/dist/thirdweb-dev-wallets-evm-wallets-metamask.esm.js';
export { RainbowWallet } from '../wallets/rainbow-wallet/dist/thirdweb-dev-wallets-evm-wallets-rainbow-wallet.esm.js';
export { WalletConnect } from '../wallets/wallet-connect/dist/thirdweb-dev-wallets-evm-wallets-wallet-connect.esm.js';
export { WalletConnectV1 } from '../wallets/wallet-connect-v1/dist/thirdweb-dev-wallets-evm-wallets-wallet-connect-v1.esm.js';
export { SafeWallet } from '../wallets/safe/dist/thirdweb-dev-wallets-evm-wallets-safe.esm.js';
export { MagicLink } from '../wallets/magic/dist/thirdweb-dev-wallets-evm-wallets-magic.esm.js';
export { S as SmartWallet, g as getAllSigners, b as getAllSmartWallets, c as getSmartWalletAddress, i as isSmartWalletDeployed } from '../../dist/smart-wallet-917da10c.esm.js';
export { EthersWallet } from '../wallets/ethers/dist/thirdweb-dev-wallets-evm-wallets-ethers.esm.js';
export { PrivateKeyWallet } from '../wallets/private-key/dist/thirdweb-dev-wallets-evm-wallets-private-key.esm.js';
export { ZerionWallet } from '../wallets/zerion/dist/thirdweb-dev-wallets-evm-wallets-zerion.esm.js';
export { TrustWallet } from '../wallets/trust/dist/thirdweb-dev-wallets-evm-wallets-trust.esm.js';
export { BloctoWallet } from '../wallets/blocto/dist/thirdweb-dev-wallets-evm-wallets-blocto.esm.js';
export { a as assertWindowEthereum } from '../../dist/assertWindowEthereum-7e869013.esm.js';
export { g as getInjectedMetamaskProvider } from '../../dist/getInjectedMetamaskProvider-d7dbfb61.esm.js';
export { g as getInjectedRainbowProvider } from '../../dist/getInjectedRainbowProvider-9357ff18.esm.js';
export { W as WagmiConnector } from '../../dist/WagmiConnector-6011bbb1.esm.js';
export { A as AddChainError, C as ChainNotConfiguredError, P as ProviderRpcError, S as SwitchChainError, U as UserRejectedRequestError } from '../../dist/errors-b9032b4e.esm.js';
export { n as normalizeChainId } from '../../dist/normalizeChainId-dd5a6036.esm.js';
export { S as SafeSupportedChainsSet } from '../../dist/constants-8444c0b0.esm.js';
import '../../dist/classPrivateMethodGet-71fe23d8.esm.js';
import '../../dist/classPrivateFieldSet-e25e1cec.esm.js';
import '../wallets/abstract/dist/thirdweb-dev-wallets-evm-wallets-abstract.esm.js';
import '../../dist/defineProperty-e24c82ea.esm.js';
import 'ethers';
import 'eventemitter3';
import '@thirdweb-dev/sdk';
import '@thirdweb-dev/chains';
import 'buffer';
import '@walletconnect/core';
import '@walletconnect/web3wallet';
import '@walletconnect/jsonrpc-utils';
