// @thirdweb-dev/wallets/evm/dist/thirdweb-dev-wallets-evm.browser.esm.js
export { E as EIP155_SIGNING_METHODS } from '../../dist/wc-c6a6a61c.browser.esm.js';
export { w as walletIds } from '../../dist/walletIds-80704ff6.browser.esm.js';
export { a as AbstractClientWallet, D as DEFAULT_DAPP_META } from '../../dist/base-2087ffde.browser.esm.js';
export { C as Connector, W as WagmiAdapter } from '../../dist/connector-05689d68.browser.esm.js';
export { PaperWallet } from '../wallets/paper-wallet/dist/thirdweb-dev-wallets-evm-wallets-paper-wallet.browser.esm.js';
export { CoinbaseWallet } from '../wallets/coinbase-wallet/dist/thirdweb-dev-wallets-evm-wallets-coinbase-wallet.browser.esm.js';
export { LocalWallet, isValidPrivateKey } from '../wallets/local-wallet/dist/thirdweb-dev-wallets-evm-wallets-local-wallet.browser.esm.js';
export { FrameWallet } from '../wallets/frame/dist/thirdweb-dev-wallets-evm-wallets-frame.browser.esm.js';
export { InjectedWallet } from '../wallets/injected/dist/thirdweb-dev-wallets-evm-wallets-injected.browser.esm.js';
export { MetaMaskWallet } from '../wallets/metamask/dist/thirdweb-dev-wallets-evm-wallets-metamask.browser.esm.js';
export { RainbowWallet } from '../wallets/rainbow-wallet/dist/thirdweb-dev-wallets-evm-wallets-rainbow-wallet.browser.esm.js';
export { WalletConnect } from '../wallets/wallet-connect/dist/thirdweb-dev-wallets-evm-wallets-wallet-connect.browser.esm.js';
export { WalletConnectV1 } from '../wallets/wallet-connect-v1/dist/thirdweb-dev-wallets-evm-wallets-wallet-connect-v1.browser.esm.js';
export { SafeWallet } from '../wallets/safe/dist/thirdweb-dev-wallets-evm-wallets-safe.browser.esm.js';
export { MagicLink } from '../wallets/magic/dist/thirdweb-dev-wallets-evm-wallets-magic.browser.esm.js';
export { S as SmartWallet, g as getAllSigners, b as getAllSmartWallets, c as getSmartWalletAddress, i as isSmartWalletDeployed } from '../../dist/smart-wallet-118d31a4.browser.esm.js';
export { EthersWallet } from '../wallets/ethers/dist/thirdweb-dev-wallets-evm-wallets-ethers.browser.esm.js';
export { PrivateKeyWallet } from '../wallets/private-key/dist/thirdweb-dev-wallets-evm-wallets-private-key.browser.esm.js';
export { ZerionWallet } from '../wallets/zerion/dist/thirdweb-dev-wallets-evm-wallets-zerion.browser.esm.js';
export { TrustWallet } from '../wallets/trust/dist/thirdweb-dev-wallets-evm-wallets-trust.browser.esm.js';
export { BloctoWallet } from '../wallets/blocto/dist/thirdweb-dev-wallets-evm-wallets-blocto.browser.esm.js';
export { a as assertWindowEthereum } from '../../dist/assertWindowEthereum-b48c1686.browser.esm.js';
export { g as getInjectedMetamaskProvider } from '../../dist/getInjectedMetamaskProvider-2a422f86.browser.esm.js';
export { g as getInjectedRainbowProvider } from '../../dist/getInjectedRainbowProvider-33485100.browser.esm.js';
export { W as WagmiConnector } from '../../dist/WagmiConnector-2f14002d.browser.esm.js';
export { A as AddChainError, C as ChainNotConfiguredError, P as ProviderRpcError, S as SwitchChainError, U as UserRejectedRequestError } from '../../dist/errors-d961f852.browser.esm.js';
export { n as normalizeChainId } from '../../dist/normalizeChainId-e4cc0175.browser.esm.js';
export { S as SafeSupportedChainsSet } from '../../dist/constants-9cc70a7d.browser.esm.js';
import '../../dist/classPrivateMethodGet-ea199cc3.browser.esm.js';
import '../../dist/classPrivateFieldSet-a5db7c83.browser.esm.js';
import '../wallets/abstract/dist/thirdweb-dev-wallets-evm-wallets-abstract.browser.esm.js';
import '../../dist/defineProperty-c8ecdc07.browser.esm.js';
import 'ethers';
import 'eventemitter3';
import '@thirdweb-dev/sdk';
import '@thirdweb-dev/chains';
import 'buffer';
import '@walletconnect/core';
import '@walletconnect/web3wallet';
import '@walletconnect/jsonrpc-utils';
